google-cloud-core/src/main/java/com/google/cloud/Identity.java

```diff
@@ -280,10 +280,12 @@ public String strValue() {
   public static Identity valueOf(String identityStr) {
     String[] info = identityStr.split(":");
     Type type = Type.valueOf(CaseFormat.LOWER_CAMEL.to(CaseFormat.UPPER_UNDERSCORE, info[0]));
-    if (type.equals(Type.ALL_AUTHENTICATED_USERS) || type.equals(Type.ALL_USERS)) {
+    if (info.length == 1) {
```

Comment:
Yes it does. I don't want to be very restrictive with server results though, because I can't predict what kind of things it will do. Users who want to pass in an Identity should be using the static factory methods anyway, which they should be drawn to because they are nice and descriptive.